src/metaopt/io/database/mongodb.py

```diff
@@ -19,95 +20,121 @@
 
 
+
 class MongoDB(AbstractDB):
     def __init__(self, dbhost='localhost', dbname='metaopt', dbport=27017, \
                                             username=None, password=None):
         self.dbhost = str(dbhost)
         self.dbname = str(dbname)
         self.username = str(username)
         self.password = str(password)
-
-        # we might have a different port for the database port
-        # so here we do a check if the port is there
-
         self.dbport = str(dbport)
 
         # set the connection string
         # this should be mongodb://mongodb0.example.net:27017
         # according to MongoDB documentation
         # https://docs.mongodb.com/getting-started/python/client/
 
-        self.uri = 'mongodb://' + self.dbhost + ':' + self.dbport
-        #self.uri = "mongodb://%s:%s@%s:%s" % (quote_plus(self.username), \
-        #                                      quote_plus(self.password), \
-        #                                     self.dbhost, self.dbport)
         # try to initialize connection
         # setting this as a function allows to make sure reconnection
         # is properly done in the case of an unreliable network
 
         self.conn = self.initiate_connection()
         self.db = self.conn[self.dbname]
 
+        if __debug__:
+            self.logger = logging.getLogger('database')
+            self.fh = logging.FileHandler('database.log')
+            self.fh.setLevel(logging.DEBUG)
+            formatter = logging.Formatter('%(asctime)s - %(name)s -\
+                                        %(levelname)s - %(message)s')
+            self.fh.setFormatter(formatter)
+            self.logger.addHandler(self.fh)
+
+    @property
+    def uri(self):
+        return self._uri
+
+    @uri.setter
+    def uri(self):
+        try:
+            # TODO: parse URI value
+            # leave that for now
+            # self.uri = 'mongodb://' + self.dbhost + ':' + self.dbport
+            uri = "mongodb://%s:%s@%s:%s" % (quote_plus(self._username), \
+                                                  quote_plus(self._password), \
+                                                 self._dbhost, self._dbport)
+            self._uri = uri
+        except:
+            raise ValueError("Incorrect values inserted for URI")
+
     def initiate_connection(self):
         try:
             conn = MongoClient(self.uri)
             self.conn = conn
         except pymongo.errors.ConnectionFailure:
             raise Exception("Could not connect to server")
+            self.logger.info("Could not connect to server")
         # return connection string for checking purposes
         return conn
 
     def check_connection(self):
         if self.db:
-            #print("Connection Successful!")
+            self.logger.info("Connection Successful!")
             return True
         else:
-            #print("Please check your connection")
+            self.logger.info("Please check your connection")
```

Comment:
It seems weird to me to print logging info inside the check_connection like this. If False, I guess the code calling the function might raise an error or something. Why should the check_connection print logging info then? Maybe that would be better to set them as debug. I don't know. @tsirif What do you think?